backend/src/apiserver/template/v2_template.go

```diff
@@ -217,3 +271,16 @@ func (t *V2Spec) RunWorkflow(modelRun *model.Run, options RunWorkflowOptions) (u
 	executionSpec.SetPodMetadataLabels(util.LabelKeyWorkflowRunId, options.RunId)
 	return executionSpec, nil
 }
+
+func isKubernetesExecutorConfig(template []byte) bool {
```

Comment:
What is the purpose of making sure the `kubernetes` exists? As we allow pipeline spec without platform spec here, we can probably let it pass with a non-kubernetes platform spec. WDYT?

Reply:
We don't need to make sure `kubernetes` exists, but we do need to prevent mis-recognize `other` platform specs as `kubernetes` spec. The current change in the PR would treat any platform spec as `kubernetes` spec.

Reply:
A question that may be unrelated: should we keep all platforms' configs in memory, or only pick out the k8s related one?

Reply:
I added a check for the key "Kubernetes".